pkg/registry/singleton.go

```diff
@@ -281,24 +280,60 @@ func (c *localSingletonConfigurator) configureHostsFile(hosts []net.IP) error {
 	return c.infraDriver.Execute(hosts, f)
 }
 
-func (c *localSingletonConfigurator) configureKubeletAuthInfo(hosts []net.IP) error {
+func (c *localSingletonConfigurator) configureAccessCredential(hosts []net.IP) error {
 	var (
-		username = c.Auth.Username
-		password = c.Auth.Username
-		endpoint = c.Domain + ":" + strconv.Itoa(c.Port)
+		username        = c.Auth.Username
+		password        = c.Auth.Password
+		endpoint        = c.Domain + ":" + strconv.Itoa(c.Port)
+		tmpAuthFilePath = "/tmp/config.json"
+		// todo we need this config file when kubelet pull images from registry. while, we could optimize the logic here.
+		remoteKubeletAuthFilePath = "/var/lib/kubelet/config.json"
 	)
 
 	if username == "" || password == "" {
 		return nil
 	}
-	// todo use sdk to login instead of shell cmd
-	configAuthCmd := fmt.Sprintf("nerdctl login -u %s -p %s %s && mkdir -p /var/lib/kubelet && cp /root/.docker/config.json /var/lib/kubelet",
-		username, password, endpoint)
 
+	err := auth.Login(context.TODO(),
+		nil,
+		&auth.LoginOptions{
+			AuthFile:           tmpAuthFilePath,
+			Password:           password,
+			Username:           username,
+			Stdout:             os.Stdout,
+			AcceptRepositories: true,
+		},
+		[]string{endpoint})
+
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		err = os.Remove(tmpAuthFilePath)
+		if err != nil {
+			logrus.Debugf("failed to remove tmp registry auth file:%s", tmpAuthFilePath)
+		}
+	}()
+
+	err = c.copy2RemoteHosts(tmpAuthFilePath, c.containerRuntimeInfo.ConfigFilePath, hosts)
+	if err != nil {
+		return err
+	}
+
+	err = c.copy2RemoteHosts(tmpAuthFilePath, remoteKubeletAuthFilePath, hosts)
```

Comment:
mkdir /var/lib/kubelet firstly, because registry is before kube-runtime

Reply:
copy2RemoteHosts will mkdir,  if remote dir is not exist .